storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3MultiPartOutputStream.java

```diff
@@ -74,7 +83,8 @@ public S3MultiPartOutputStream(final String bucketName,
         this.partSize = partSize;
         this.partBuffer = ByteBuffer.allocate(partSize);
         final CreateMultipartUploadRequest initialRequest = CreateMultipartUploadRequest.builder().bucket(bucketName)
-            .key(key.value()).build();
+                .storageClass(storageClass)
+                .key(key.value()).build();
```

Comment:
Should we add a test on S3MultiPartOutputStreamTest to validate that the initialRequest is getting the right storage class via Mockito?